packages/medusa/src/api/auth/[actor_type]/[auth_provider]/callback/route.ts

```diff
@@ -14,10 +14,11 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
     ContainerRegistrationKeys.CONFIG_MODULE
   )
 
-  const authMethods = (config.projectConfig?.http as any)?.authMethods ?? {}
+  const authMethodsPerActor =
+    config.projectConfig?.http?.authMethodsPerActor ?? {}
   // Not having the config defined would allow for all auth providers for the particular actor.
-  if (authMethods[actor_type]) {
-    if (!authMethods[actor_type].includes(auth_provider)) {
+  if (authMethodsPerActor[actor_type]) {
+    if (!authMethodsPerActor[actor_type].includes(auth_provider)) {
```

Comment:
```suggestion
    if (!authMethodsPerActor[actor_type]?.includes(auth_provider)) {
```

Reply:
otherwise if the auth actor provided is not configured, it will go through, wdyt?

Reply:
@adrien2p that's what we want to achieve, if it is not defined it should allow any login method by default

Reply:
ok got it then maybe we can reduce the number of if and do
```ts
if (authMethodsPerActor[actor_type] && !authMethodsPerActor[actor_type].includes(auth_provider)) {
```

wdyt?

Reply:
@adrien2p I found this slightly more readable as it is not so long, but it doesn't really matter IMO

Reply:
yeah it doesnt really, But I know that usually you are trying to prevent those from happening elsewhere so I thoughtI would mention it, but it doesnt change the behaviour so fine to keep it